Windows 빌드 실패: pluginClass 가 C API 패턴과 불일치 → generated_plugin_registrant 가 존재하지 않는 헤더/함수 참조

## 문제

`flutter_image_conversion: ^2.2.0` 을 의존성으로 사용하는 Flutter 프로젝트에서 `flutter build windows` 실행 시 `generated_plugin_registrant.cc` 가 존재하지 않는 헤더를 포함하려 해 빌드가 실패합니다:

```
windows/flutter/generated_plugin_registrant.cc(12,10): error C1083:
  포함 파일을 열 수 없습니다.
  'flutter_image_conversion/flutter_image_conversion_plugin.h':
  No such file or directory
```

자매 패키지 `flutter_device_platform_id` 에서도 동일한 증상이 발견되어 [v1.0.3](https://pub.dev/packages/flutter_device_platform_id/versions/1.0.3) 에서 동일한 방식으로 해결된 바 있습니다 ([cyberprophet/flutter-device-unique-id#8](https://github.com/cyberprophet/flutter-device-unique-id/issues/8)).

## 원인

플러그인의 Windows 측은 **C API 패턴**으로 구현되어 있습니다:

- 공개 헤더: `windows/include/flutter_image_conversion/flutter_image_conversion_plugin_c_api.h`
  - 노출 함수: `FlutterImageConversionPluginCApiRegisterWithRegistrar`
- 내부 헤더: `windows/flutter_image_conversion_plugin.h` (`include/` 외부 — 다운스트림에 노출되지 않음)
- 구현: `windows/flutter_image_conversion_plugin.cpp` 의 L51–56 에서 `FlutterImageConversionPluginCApiRegisterWithRegistrar` 정의

`windows/CMakeLists.txt:21-22` 는 `include/` 만 `INTERFACE` 로 노출하므로, 내부 헤더 `flutter_image_conversion_plugin.h` 는 다운스트림 프로젝트에서 참조 불가입니다.

반면 `pubspec.yaml` 은:

```yaml
flutter:
  plugin:
    platforms:
      windows:
        pluginClass: FlutterImageConversionPlugin
```

`pluginClass: FlutterImageConversionPlugin` 은 **직접 클래스 패턴**으로 인식되어 Flutter 툴이:

1. `<flutter_image_conversion/flutter_image_conversion_plugin.h>` 를 include 하고 (→ 존재하지 않음)
2. `FlutterImageConversionPluginRegisterWithRegistrar(...)` 를 호출합니다 (→ 정의되지 않음. 실제 노출된 심볼은 `...CApiRegisterWithRegistrar`)

즉 구현(C API 패턴)과 pubspec 선언(직접 클래스 패턴)이 불일치합니다.

## 제안 수정

C API 패턴을 유지하는 전제로 `pubspec.yaml` 의 Windows `pluginClass` 를 `Plugin` → `PluginCApi` 로 교체. 소스 코드 변경 없이 한 줄 수정으로 해결됩니다.

```diff
 flutter:
   plugin:
     platforms:
       ...
       windows:
-        pluginClass: FlutterImageConversionPlugin
+        pluginClass: FlutterImageConversionPluginCApi
```

Flutter 툴은 `pluginClass` 가 `CApi` 로 끝나면 자동으로 include 를 `..._plugin_c_api.h` 로, register 함수를 `...CApiRegisterWithRegistrar` 로 생성하므로, 이미 구현되어 있는 공개 API 와 일치하게 됩니다.

## 재현

```yaml
# consumer pubspec
dependencies:
  flutter_image_conversion: ^2.2.0
```

```
flutter clean && flutter pub get && flutter build windows --debug
```

## 환경

- Flutter 3.41.2 (stable)
- Visual Studio 2022 / MSVC v143
- Windows 11 Pro 24H2
- `flutter_image_conversion: 2.2.0`

---

Requested by P6 creative-client (Windows 배포 확장 작업 중 발견).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,18 @@
+## 2.2.1
+
+**Patch Release: Fix Windows Build Failure**
+
+### Bug Fixes
+- Fixed `flutter build windows` failure in downstream projects depending on `flutter_image_conversion`
+  - Windows plugin is implemented with the C API pattern (exposes `FlutterImageConversionPluginCApiRegisterWithRegistrar` via `include/flutter_image_conversion/flutter_image_conversion_plugin_c_api.h`), but `pubspec.yaml` declared `pluginClass: FlutterImageConversionPlugin`, causing Flutter's tool to generate `generated_plugin_registrant.cc` that referenced a non-existent header and an undefined register function.
+  - Changed Windows `pluginClass` from `FlutterImageConversionPlugin` to `FlutterImageConversionPluginCApi` so the generated registrant correctly includes `flutter_image_conversion_plugin_c_api.h` and calls `FlutterImageConversionPluginCApiRegisterWithRegistrar`.
+
+### Notes
+- No source code changes; `pubspec.yaml` manifest fix only.
+- Fixes #6.
+
+---
+
 ## 2.2.0
 
 **Feature: WASM Compatibility**

--- a/example/windows/flutter/generated_plugin_registrant.cc
+++ b/example/windows/flutter/generated_plugin_registrant.cc
@@ -6,9 +6,9 @@
 
 #include "generated_plugin_registrant.h"
 
-#include <flutter_image_conversion/flutter_image_conversion_plugin.h>
+#include <flutter_image_conversion/flutter_image_conversion_plugin_c_api.h>
 
 void RegisterPlugins(flutter::PluginRegistry* registry) {
-  FlutterImageConversionPluginRegisterWithRegistrar(
-      registry->GetRegistrarForPlugin("FlutterImageConversionPlugin"));
+  FlutterImageConversionPluginCApiRegisterWithRegistrar(
+      registry->GetRegistrarForPlugin("FlutterImageConversionPluginCApi"));
 }

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -3,7 +3,7 @@ description: >
   A lightweight Flutter plugin that enables conversion of HEIC images to JPEG format on iOS, macOS, and Web.
   Supports Windows and Android with graceful fallback. Ideal for apps targeting image uploads, social sharing, or cross-platform compatibility.
 
-version: 2.2.0
+version: 2.2.1
 homepage: https://github.com/cyberprophet/flutter-image-conversion
 
 environment:
@@ -34,7 +34,7 @@ flutter:
       macos:
         pluginClass: FlutterImageConversionPlugin
       windows:
-        pluginClass: FlutterImageConversionPlugin
+        pluginClass: FlutterImageConversionPluginCApi
       web:
         pluginClass: FlutterImageConversionWeb
         fileName: flutter_image_conversion_web.dart

